types/webpack-env/index.d.ts

```diff
@@ -103,21 +104,21 @@ declare namespace __WebpackModuleApi {
          * The data will be available at module.hot.data on the new module.
          * @param callback
          */
-        dispose<T>(callback: (data: T) => void): void;
+        dispose<T = any>(callback: (data: T) => void): void;
```

Comment:
As Daniel said, defaulting to `any` is equivalent to making the function non-generic. It makes no sense IMHO. I don't want to sound abrasive, but I think this PR goes against best practices. There is a reason why TypeScript falls back to `{}` instead of `any`. Please think of type safety and `strict` option as well.

As a personal opinion, `any` is nothing more than an escape hatch. 9 cases out of 10 using `any` is the wrong answer. In a d.ts file it's worse, because it propagates to the consumer as well. That said, I'd rather force the consumer to pass `any` as a type argument explicitly, than implicitly casting the result as `any` for everyone. Any serious project (no pun intended) should be using the `strict` option, so I'd treat this as a must rather than a nice to have.